modules/nixos/programs/docker/karakeep-singlefile-bridge/files/singlefile-bridge.py

```
@@ -0,0 +1,629 @@
+        if detach_existing_full_archive:
+            conn.execute(
+                "UPDATE assets SET assetType='unknown', bookmarkId=NULL "
+                "WHERE bookmarkId=? AND assetType='linkFullPageArchive' AND id<>?",
+                (bookmark_id, asset_id),
+            )
+        cur = conn.execute(
+            "UPDATE assets SET bookmarkId=?, assetType='linkFullPageArchive' WHERE id=?",
+            (bookmark_id, asset_id),
+        )
+        if cur.rowcount != 1:
+            raise RuntimeError("failed to link uploaded asset as fullPageArchive")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)